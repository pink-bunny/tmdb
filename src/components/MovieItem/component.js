import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Card } from 'antd';

const Movie = ({
  actions,
  title,
  overview,
  poster,
  id
}) => {
  const imagePath = poster ? `url(https://image.tmdb.org/t/p/w300_and_h450_bestv2/${poster})` : 'none';

  return (
    <Card
      className="movie-item top-margin"
      cover={(
        <Link
          className="movie-item__img"
          to={`movie/${id}`}
          style={{ backgroundImage: imagePath }}
        />
      )}
      actions={actions}
    >
      <Card.Meta
        title={title}
        description={overview}
      />
    </Card>
  );
};

Movie.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.node),
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  poster: PropTypes.string,
  id: PropTypes.number.isRequired
};

Movie.defaultProps = {
  actions: [],
  poster: ''
};

export default Movie;
