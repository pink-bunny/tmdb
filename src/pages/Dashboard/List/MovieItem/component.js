import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'antd';

const Movie = ({
  actions,
  title,
  overview,
  poster
}) => (
  <Card
    hoverable
    cover={(
      <img
        alt={title}
        src={`https://image.tmdb.org/t/p/w300_and_h450_bestv2/${poster}`}
      />
    )}
    className="top-margin"
    actions={actions}
  >
    <Card.Meta
      title={title}
      description={overview}
    />
  </Card>
);

Movie.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.node),
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired
};

Movie.defaultProps = {
  actions: []
};

export default Movie;
