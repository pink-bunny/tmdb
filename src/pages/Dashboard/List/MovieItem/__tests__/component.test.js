import React from 'react';
import { shallow } from 'enzyme';

import MovieItem from '../component';

describe('MovieItem component matches snapshot', () => {
  const requiredProps = {
    title: 'Terminator: Dark Fate',
    overview: 'Decades after Sarah Connor prevented Judgment Day.'
  };

  it('with required props', () => {
    const wrapper = shallow(<MovieItem {...requiredProps} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('with poster', () => {
    const props = {
      ...requiredProps,
      poster: '/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg' /* eslint-disable-line */
    };
    const wrapper = shallow(<MovieItem {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
