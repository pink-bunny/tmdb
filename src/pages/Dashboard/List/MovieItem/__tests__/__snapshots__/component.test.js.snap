// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem component matches snapshot with poster 1`] = `
<Card
  actions={Array []}
  className="top-margin"
  cover={
    <img
      alt="Terminator: Dark Fate"
      src="https://image.tmdb.org/t/p/w300_and_h450_bestv2//vqzNJRH4YyquRiWxCCOH0aXggHI.jpg"
    />
  }
  hoverable={true}
>
  <Meta
    description="Decades after Sarah Connor prevented Judgment Day."
    title="Terminator: Dark Fate"
  />
</Card>
`;

exports[`MovieItem component matches snapshot with required props 1`] = `
<Card
  actions={Array []}
  className="top-margin"
  cover={null}
  hoverable={true}
>
  <Meta
    description="Decades after Sarah Connor prevented Judgment Day."
    title="Terminator: Dark Fate"
  />
</Card>
`;
